xbmc/pvr/PVRManager.cpp

```diff
@@ -113,23 +113,38 @@ CPVRManager::~CPVRManager(void)
 
 void CPVRManager::Announce(AnnouncementFlag flag, const char *sender, const char *message, const CVariant &data)
 {
-  if (!IsStarted() || (flag & (System)) == 0)
+  if (!IsStarted())
     return;
 
-  if (strcmp(message, "OnWake") == 0)
+  if ((flag & (ANNOUNCEMENT::System)))
+  {
+    if (strcmp(message, "OnSleep") == 0)
```

Comment:
this is not reliable. there is no guarantee that async announce messages get delivered before system gets down. you may get this message on wakeup.
that why I said the the IAnnounce interface was abused for those purposes. System behaviour should not depend on messages coming through announce.

Better do something like g_PVRManager.OnSleep/OnWake/etc